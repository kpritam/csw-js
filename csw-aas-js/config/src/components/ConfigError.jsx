import React from 'react'

const ConfigError = () => <div>Please login to proceed</div>

export default ConfigError
