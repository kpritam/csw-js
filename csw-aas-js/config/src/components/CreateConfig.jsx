import React from 'react'
import IOOperationComponent from './IOOperationComponent'
import { sPost } from './Client'
import { Consumer } from 'csw-aas-js'

class CreateConfig extends React.Component {
  state = { response: null, token: null, fileContent: '' }

  callBack = res => {
    this.setState({ response: res })
  }

  createConfig = (input, token) => {
    sPost(
      `http://localhost:5000/config/${input}?comment="Sample commit message"`,
      this.callBack,
      token,
      this.state.fileContent,
    )
  }

  updateFileContent = event => {
    this.setState({
      fileContent: event.target.value,
    })
  }

  render() {
    return (
      <Consumer>
        {({ auth }) => {
          return (
            <div className='card-panel hoverable'>
              <IOOperationComponent
                token={auth.token}
                componentNameProp='Create Config'
                operation='Create Config'
                output={this.state.response}
                api={this.createConfig}
              />
              <div className='card-panel hoverable'>
                File Content
                <span>
                  <textarea
                    value={this.state.fileContent}
                    onChange={this.updateFileContent}
                  />
                </span>
              </div>
            </div>
          )
        }}
      </Consumer>
    )
  }
}

export default CreateConfig
