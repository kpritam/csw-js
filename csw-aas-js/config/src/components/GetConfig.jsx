import React from 'react'
import IOOperationComponent from './IOOperationComponent'

class GetConfig extends React.Component {
  state = { response: null }

  downloadURI = uri => {
    let link = document.createElement('a')
    link.href = uri
    document.body.appendChild(link)
    link.click()
    document.body.removeChild(link)
  }

  getConfig = async input => {
    console.log(input)
    this.downloadURI(`http://localhost:5000/config/${input}`)
  }

  render() {
    return (
      <IOOperationComponent
        componentNameProp='Get Config'
        operation='Get'
        output={this.state.response}
        api={this.getConfig}
      />
    )
  }
}

export default GetConfig
