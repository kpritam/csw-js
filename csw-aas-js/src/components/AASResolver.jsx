import { Config } from '../config/configs'

/*
 * URL constant which contains location service end point for resolving AAS server
 */
const URL = `${Config['location-server-url']}/location/resolve/${
  Config['AAS-server-name']
}?within=5seconds`

/**
 * Utility method responsible for resolving AAS server using location service
 * @param url default param which contains location service end point for resolving
 * AAS server
 * @returns AAS url if resolved from location service else return null
 */
export const resolveAAS = async function(url = URL) {
  const response = await window.fetch(url)
  if (response.status === 200) {
    const a = await response.json()
    return a.uri
  }
  return null
}
