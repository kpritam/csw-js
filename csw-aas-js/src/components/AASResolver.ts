import { Config } from '../config/configs'

/**
 * Utility method responsible for resolving AAS server using location service
 *
 * @returns AAS url if resolved from location service else return null
 */
export const resolveAAS: () => Promise<string | null> = async () => {
  // URL constant which contains location service end point for resolving AAS server
  const URL = `${Config['location-server-url']}/location/resolve/${
    Config['AAS-server-name']
  }?within=5seconds`

  const response = await window.fetch(URL)
  if (response.status === 200) {
    const jsonResponse = await response.json()
    return jsonResponse.uri
  }
  return null
}
