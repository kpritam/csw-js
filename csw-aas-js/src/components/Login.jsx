import React from 'react'
import PropTypes from 'prop-types'
import WithContext from './context/WithContext'

/**
 * React component which renders Login button.
 */
export class Login extends React.Component {
  render() {
    return (
      <button
        onClick={async () => {
          // Call to this.props.login method is responsible for resolving and instantiating AAS server
          await this.props.login()
        }}>
        Login
      </button>
    )
  }
}

Login.propTypes = {
  // WithContext utility provides login method as a prop.
  login: PropTypes.func,
}

export default WithContext(Login)
