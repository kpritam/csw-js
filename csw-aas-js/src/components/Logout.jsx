import React from 'react'
import PropTypes from 'prop-types'
import WithContext from './context/WithContext'

/**
 * React component which renders Logout button.
 */
export class Logout extends React.Component {
  render() {
    return (
      <button
        onClick={async () => {
          // Call to this.props.logout method is responsible for logging out.
          await this.props.logout()
        }}>
        Logout
      </button>
    )
  }
}

Logout.propTypes = {
  // WithContext utility provides logout method as a prop.
  logout: PropTypes.func,
}

export default WithContext(Logout)
