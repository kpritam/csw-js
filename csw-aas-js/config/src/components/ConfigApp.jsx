import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import { ClientRole, AuthContextProvider } from 'csw-aas-js'
import NavComponent from './NavComponent'
import { AppConfig } from '../config/AppConfig'
import WriteConfig from './CreateConfig'
import ConfigError from './ConfigError'
import ListConfig from './ListConfig'
import GetConfig from './GetConfig'

const ConfigApp = () => {
  return (
    <div className='row card col s12 m7'>
      <AuthContextProvider config={AppConfig}>
        <BrowserRouter>
          <div>
            <NavComponent />
          </div>
        </BrowserRouter>
        <ListConfig />
        <GetConfig />
        <ClientRole
          clientRole='admin'
          client='csw-config-server'
          error={<ConfigError />}>
          <WriteConfig />
        </ClientRole>
      </AuthContextProvider>
    </div>
  )
}

export default ConfigApp
