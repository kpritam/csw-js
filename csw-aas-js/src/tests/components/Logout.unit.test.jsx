import React from 'react'
import { Logout } from '../../components/Logout'
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

// DEOPSCSW-630 - Javascript adapter for AAS
// DEOPSCSW-631 - React layer for javascript adapter for AAS
describe('<Logout />', () => {
  Enzyme.configure({ adapter: new Adapter() })

  it('should call logout', async () => {
    const props = {
      logout: jest.fn(),
    }

    const wrapper = await mount(<Logout {...props} />)

    wrapper.find('button').simulate('click')

    expect(wrapper.props().logout).toHaveBeenCalled()

    wrapper.unmount()
  })

  it('should render logout', () => {
    const props = {
      logout: jest.fn(),
    }

    const logout = renderer.create(<Logout {...props} />).toJSON()
    expect(logout).toMatchSnapshot()
  })
})
