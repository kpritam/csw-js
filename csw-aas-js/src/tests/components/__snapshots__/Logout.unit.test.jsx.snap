// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout /> should render logout 1`] = `
<button
  onClick={[Function]}
>
  Logout
</button>
`;
